torch/_inductor/pcache.py

```diff
@@ -0,0 +1,315 @@
+        - For asynchronous and thread-safe cache, see `AsyncCache`.
+
+    Methods:
+        get(key): Retrieve a value by key.
+        insert(key, value): Insert a value if the key does not already exist.
+    """
+
+    def get(self: Self, key: Key) -> Value | None:
+        """
+        Retrieve the value associated with the given key from the cache.
+
+        Args:
+            key (Key): The key used to query the cache.
+
+        Returns:
+            Value | None: The value associated with the key, or None if not found.
+
+        Raises:
+            NotImplementedError: If not implemented by a subclass.
+        """
+        raise NotImplementedError
+
+    def insert(self: Self, key: Key, value: Value) -> bool:
+        """
+        Store the given value in the cache with the associated key if the key does
+        not already exist in the cache, otherwise do nothing.
+
+        Args:
+            key (Key): The key to associate with the value.
+            value (Value): The value to be stored in the cache.
+
+        Returns:
+            bool: True if the value was stored successfully, False if the key already exists.
+
+        Raises:
+            NotImplementedError: If not implemented by a subclass.
+        """
+        raise NotImplementedError
+
+
+class InMemoryCache(Cache[str, bytes]):
+    """
+    In-memory cache implementation.
+
+    Stores cache data in a dictionary for fast lookups and insertions.
+    Not thread-safe.
+    """
+
+    def __init__(self: Self) -> None:
+        """
+        Initialize the in-memory cache.
+        """
+        self._cache: dict[str, bytes] = {}
+
+    def get(self: Self, key: str) -> bytes | None:
+        """
+        Retrieve the value associated with the given key from the cache.
+
+        Args:
+            key (str): The key used to query the cache.
+
+        Returns:
+            bytes | None: The value associated with the key, or None if not found.
+        """
+        return self._cache.get(key, None)
+
+    def insert(self: Self, key: str, value: bytes) -> bool:
+        """
+        Store the given value in the cache with the associated key if the key does
+        not already exist in the cache, otherwise do nothing.
+
+        Args:
+            key (str): The key to associate with the value.
+            value (bytes): The value to be stored in the cache.
+
+        Returns:
+            bool: True if the value was stored successfully, False if the key already exists.
+        """
+        if key in self._cache:
+            return False
+        self._cache[key] = value
+        return True
+
+    @classmethod
+    def from_env_var(cls, env_var: str) -> Self:
+        """
+        Create a new in-memory cache instance from an environment variable.
+
+        The environment variable should contain key-value pairs separated by ';',
+        with each pair formatted as 'key,value'. The value should be a string
+        representation of bytes (e.g., b'...').
```

Comment:
So the values must be literally in the form `b'abc'`, no double-quotes, no pure ascii? That seems kind of restrictive?